app/routes.py

```diff
@@ -1 +1,291 @@
+        return {"task":{
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                    }
+                }
+    else:
+        return {"task":{
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False
+                    }
+                }
+
+#####GOAL#####
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = validate_goal(goal_id)
+    return {"goal":{
+                "id": goal.goal_id,
+                "title": goal.title
+                }
+            }
+#####TASK######   
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_task(task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    response = {"task":{
+                "description": task.description,
+                "id": task.task_id,
+                "is_complete": False,
+                "title": task.title 
+            }}
+    return make_response(jsonify(response), 200)
+####GOAL######
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_goal(goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    response = {"goal":{
+                "id": goal.goal_id,
+                "title": goal.title 
+            }}
+    return make_response(jsonify(response), 200)
+
+#####TASK#######
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_incompleted_task(task_id):
+    task = validate_task(task_id)
+    request_body = request.get_json()
+    task.completed_at = datetime.now() 
+    if task.completed_at == None: 
+        status = False
+    else:
+        status = True
+    db.session.commit()
+    post_slack_message(f"Someone just completed the task {task.title}.")
+    response = {"task":{
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": status,
+    }}
+    return make_response(jsonify(response), 200)
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete_on_completed_task(task_id):
+    task = validate_task(task_id)
+    request_body = request.get_json()
+    task.completed_at = None
+    if task.completed_at == None: 
+        status = False
+    else:
+        status = True
+    db.session.commit()
+    response = {"task":{
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": status,
+    }}
```

Comment:
Perhaps we can find a way to calculate `is_complete` based on the `completed_at` attribute rather than creating a new variable to hold the boolean value.